### PR TITLE
Core/Scripts - Karazhan Init

### PR DESCRIPTION
**Pull request by FALL1N1**
This is a WIP so place the WIP tag and leave it as-is, also do not merge with master please, I will do it when it's ready.

## CHANGES PROPOSED:
-  Script the event I guess?

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/779

## TESTS PERFORMED:
- None :(

## HOW TO TEST THE CHANGES:

https://www.youtube.com/watch?v=6hcmsBkZvTY
Tele to the specific boss/encounter: .go xyz -11093 -1847 221 532

## KNOWN ISSUES AND TODO LIST:

it's WIP

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FALL1N1** [comment]
ready to be tested

**Comment by Helias** [comment]
Chess Event fixed? good :+1: 

@FALL1N1 fix the build errors that the pipeline shows pls

**Comment by FALL1N1** [comment]
@Helias, it's not fixed yet, there aren't conditions for the spells.

**Comment by Krath** [comment]
Is this ready to be tested? I applied this PR to my local source, and once I clicked on Warchief Blackhand he started moving out of map.

By the way, I got this error when importing : Error Code: 1054 - Unknown column 'spawndist' in 'field list'

File: data/sql/updates/pending_db_world/0000_00_00_00_karazhan_chess.sql [hunk 1]
```diff
@@ -0,0 +1,455 @@
+update creature_template set scriptname = "npc_chesspiece" where entry in (17469,17211,21748,21664,21750,21683,21747,21682,21726,21160,21752,21684);
+update creature_template set scriptname = "npc_echo_of_medivh" where entry = 16816;
+update creature_template set scriptname = "npc_chess_move_trigger",flags_extra=flags_extra|128,modelid2=1126 where entry = 22519;
+
+DELETE FROM `spell_script_names` WHERE `spell_id`=30019;
+
+DELETE FROM `creature` WHERE `id` IN (17469,17211,21748,21664,21750,21683,21747,21682,21726,21160,21752,21684);
+INSERT INTO `creature` (`guid`,`id`,`map`,`spawnMask`,`phaseMask`,`modelid`,`equipment_id`,`position_x`,`position_y`,`position_z`,`orientation`,`spawntimesecs`,`spawndist`,`currentwaypoint`,`curhealth`,`curmana`,`MovementType`,`npcflag`,`unit_flags`,`dynamicflags`) VALUES
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/0000_00_00_00_karazhan_chess.sql:8]: There is no such thing as spawndist, from April it is wander_distance :)

**Comment by FALL1N1** [comment]
@Krath, in general it is ready, but it needs someone with 'conditions' knowlege to rework all of them, so the spells can have proper targets.

**Comment by FrancescoBorzi** [comment]
@FALL1N1 have you given a try with Keira3 when editing conditions?

**Comment by Krath** [comment]
> @Krath, in general it is ready, but it needs someone with 'conditions' knowledge to rework all of them, so the spells can have proper targets.

Did you script it or you just ported it from another core?
The event can't get completed, once you click on the King's gossip to control him, he walks outside the map and you can't do anything else

**Comment by JamesGent** [comment]
As soon as I speak to Warchief Blackhand he starts clipping through the floor and walls. I wasn't able to test any further than this, sorry. :(

**Comment by FALL1N1** [comment]
It is ported from CMangos, hence it needs accurate spell conditions as mentioned.

**Comment by Krath** [comment]
> It is ported from CMangos, hence it needs accurate spell conditions as mentioned.

Oh alright, I thought that hadn't to do with the script itself! I'm glad to help with the testing when done :)

**Comment by Kitzunu** [comment]
Conflicts @FALL1N1

**Comment by BarbzYHOOL** [comment]
what is required? DB or Core work? @FALL1N1

**Comment by FALL1N1** [comment]
> 
> 
> what is required? DB or Core work? @FALL1N1
> and it's for the chess event?

core work is 100% done as far as I know how the encounter goes, what we need now is only conditions for the spells, e.g 'Move' targets a stalker at the selected area to cast, 'Rotate' targets a nearby stalker (90*)..etc

**Comment by BarbzYHOOL** [comment]
How to complete the PR (after discussing with fall1n):

This requires to complete the conditions for the chess pieces spells (moving, attacking), it's DB work. Anyone can do it with logic and researches. Spells in general, are coded, but missing conditions (on how they should "operate")

Sample conditions: https://github.com/azerothcore/azerothcore-wotlk/blob/4df0e7a3b1da3f081359e26f2eceab10fa522cc5/data/sql/updates/pending_db_world/0000_00_00_00_karazhan_chess.sql

for example, when you get on the vehicle npcs (chess figures) there are two main spells 1) Move 2) Rotate For example, "Move" has no conditions at the moment to "point" the spell's target, e.g "Pawn" can move only 2 lines front and 1 when over half of the field. Instead, now they just move to the nearest trigger(invisible npc) they find.

Method: If i were to do that work, i would analyze the existing conditions from the PR, test in game how it's working atm then deduce and fix the rest. When i dunno what is doing what, i delete it to see what happens. Can also analyze random conditions too

Resources: https://www.azerothcore.org/wiki/conditions and keira3 can help a lot

**Pull request closed without merging**